src/bootstrap-tagsinput.js

```diff
@@ -125,16 +129,35 @@
         return;
 
       // raise beforeItemAdd arg
-      var beforeItemAddEvent = $.Event('beforeItemAdd', { item: item, cancel: false, options: options});
-      self.$element.trigger(beforeItemAddEvent);
-      if (beforeItemAddEvent.cancel)
+	  var beforeItemAddEvent = $.Event('beforeItemAdd', { item: item, cancel: false, options: options});	  
+	  
+	  self.$element.trigger(beforeItemAddEvent);	  	  
+      if (beforeItemAddEvent.cancel) 
         return;
 
       // register item in internal array and map
-      self.itemsArray.push(item);
-
+      self.itemsArray.push(item);	
+	  
+	  // read var beforeItemAddEvent with new value
+	  var item = beforeItemAddEvent.item; // Get text from event (BeforeItemAddEvent)	  
```

Comment:
so the way I want to use this feature --> I had to move  line 139 below line 142..

i.e.  self.itemsArray.push(item); needs to happen after:
var item = beforeItemAddEvent.item; // Get text from event (BeforeItemAddEvent)

does that make sense? The reason is because these tags are populating a text field which is saved... if i dont move that push below getting the text from BeforeItemAddEvent the tag displays the text returned by BeforeItemAddEvent; however, the unmodified value is stored and on subequent display of this saved textfield it just shows the unmassaged text (i..e i was using BeforeItemAddEvent to massage the user input) Im currently invesigating, as I may want to move the duplicate checking stuff below this too if possible... i.e. basically i want to take the user inpt and modify it and handle everything just like the stuff I modified it to was what the user input in the first place... if that makes any sense.